### ISSUE TITLE
❓ If frameProcessorFps is 5, shouldn't frameProcessor run 5 times per second?

### ISSUE THREAD

**Issue by Noma98** [cuvent/react-native-vision-camera#1202]
### Question

If frameProcessorFps is 5, shouldn't frameProcessor run 5 times per second? Assigning 1 to frameProcessorFps calls exactly once every second. However, if I assign a number greater than 2, it is called only once or twice a second.

Please see the code below (this is an excerpt of my project code made using reaction-native-vision-camera). The text value recognized in runOnJs is saved as ocr and the processRecognizedText is executed whenever the ocr value changes to useEffect. At this time, I guessed that the frame processor could not run as many times as I wanted per second because it took a long time to run processRecognizedText, but when I checked it, it took 0.001 seconds.

If so, I think processRecognizedText should be excuted 100 times if frameProcessorFps is set to 100, but why is it only excuted 1~2 times?
I would appreciate it if you could check if I misunderstood the function of frame processor Fps and if there was anything wrong set up! Thank you.
```
import React, { useEffect, useState } from 'react';
import 'react-native-reanimated';
import { runOnJS } from 'react-native-reanimated';
import {
  Camera,
  useCameraDevices,
  useFrameProcessor,
} from 'react-native-vision-camera';
import { scanOCR } from 'vision-camera-ocr';
(...)

const DataCaptureView = ({ navigation, route }: IProps) => {
(...)
  const [enabled, setEnabled] = useState(true);
  const [ocr, setOcr] = useState();
  const devices = useCameraDevices();
  const device = devices[cameraPosition];

  const frameProcessor = useFrameProcessor(frame => {
    'worklet';
    const data = scanOCR(frame);
    runOnJS(setOcr)(data?.result?.text);
  }, []);

  useEffect(() => {
    if (!Boolean(ocr)) {
      return;
    }
    enabled && processRecognizedText(ocr);
  }, [ocr, enabled]);

  const processRecognizedText = (text: string) => {
    (...)
  };

  return (
    <StyledView>
      {device !== undefined && hasPermission ? (
          <Camera
            style={{ flex: 1, width: '100%' }}
            frameProcessor={frameProcessor}
            device={device}
            isActive={isAppForeground && isFocused}
            frameProcessorFps={5}
          />
      ) : (
        <StyledView />
      )}
    </StyledView>
  );
};
const StyledView = styled.View`
  flex: 1;
`;
export default DataCaptureView;
```

### VisionCamera Version

2.13.5

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for similar questions [in the issues page](https://github.com/mrousavy/react-native-vision-camera/issues?q=is%3Aopen+is%3Aissue+label%3A%22❓+question%22) as well as [in the discussions page](https://github.com/mrousavy/react-native-vision-camera/discussions) and found none.

**Comment by hikmert** [comment]
I have exactly the same issue, setting frameprocessorfps to , let's say 30, and I only receive around 12-13 frames from native Android code. Any ideas?

**Comment by thomas-coldwell** [comment]
Hey everyone, I think I might have some insight onto your issues here:
- Firstly, on Android you can set the Frame Processor FPS as high as you like but internally it will be capped at the FPS CameraX decides on for its ImageAnalysis use case (in my testing this was around 15-20fps). There are ways to improve this FPS, but it requires setting up some custom capture requests with Camera2 through CameraX's interop API - maybe this is something we could think about sorting out for V3 so you can disable some camera features to improve the ImageAnalysis use case @mrousavy ?
- Secondly, if you're running this on iOS I just fixed an issue that you can see in this PR #1288 that was causing it to skip ever other frame (if the frame processor callback was able to execute in real-time)

**Comment by mrousavy** [comment]
> but it requires setting up some custom capture requests with Camera2 through CameraX's interop API

CameraX is limiting us more and more. We should talk about using Camera2 instead

**Comment by mrousavy** [comment]
Closing as this is a stale issue - this might have been fixed with the full rewrite in VisionCamera V3 (🥳) - if not, please create a new issue.
